packages/blockchain-wallet-v4-frontend/src/scenes/Nfts/Asset/index.tsx

```diff
@@ -1,36 +1,715 @@
+  font-weight: 600;
+  font-size: 14px;
+  line-height: 20px;
+  padding: 1em;
+`
+
+const CreatorOwnerAddress = styled.div`
+  font-size: 16px;
+  line-height: 150%;
+  color: #828b9e;
+  overflow: hidden;
+  padding-left: 1em;
+`
+
+const Divider = styled.div`
+  margin-bottom: 1em;
+  position: static;
+  width: 727px;
+  height: 1px;
+  left: 48px;
+  top: 51px;
+  background: #f0f2f7;
+`
+
+const DividerNoMargin = styled(Divider)`
+  margin-bottom: unset;
+  width: 100%;
+`
+
+const Description = styled.div`
+  padding: 3em 0em 3em 0em;
+  font-family: Inter, sans-serif;
+  font-style: normal;
+  font-weight: 500;
+  font-size: 16px;
+  line-height: 24px;
+  color: #677184;
+`
+
+const TraitsWrapper = styled(CurrentPriceBox)`
+  margin-top: 2em;
+  padding: 1em;
+  height: unset;
+  font-family: Inter, sans-serif;
+  font-style: normal;
+  font-weight: 600;
+  font-size: 14px;
+  color: #677184;
+`
+const TraitCell = styled.div`
+  padding: 1em;
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  gap: 10px;
+  max-width: 100%;
+`
+
+const Trait = styled.div`
+  font-family: Inter, sans-serif;
+  font-style: normal;
+  font-weight: 600;
+  font-size: 12px;
+  background: rgba(240, 242, 247, 0.4);
+  border-radius: 8px;
+`
+
+const AddressDisplay = styled.div`
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  overflow: hidden;
+  width: 90%;
+`
+
+const AdditionalDetailsWrapper = styled(TraitsWrapper)``
+
+const AdditionalDetails = styled.div`
+  padding: 1em;
+  color: #828b9e;
+`
+
+const SocialLinksWrap = styled.div`
+  display: flex;
+  flex-direction: row;
+  padding: 0.5em;
+  border-radius: 20px;
+`
+
+const SocialLinks = styled.a.attrs({
+  target: '_blank'
+})`
+  display: flex;
+  height: 1.5rem;
+  width: 1.5rem;
+  border-radius: 100%;
+  background-color: ${Color('white')} !important;
+  color: white;
+  transition: all 0.5s;
+  margin-right: 0.5rem;
+  justify-content: center;
+
+  .social-icons {
+    opacity: 0.5;
+  }
+
+  &:hover {
+    .social-icons {
+      opacity: 1;
+    }
+  }
+
+  @media only screen and (max-width: 48rem) {
+    height: 2rem;
+    width: 2rem;
+    margin-top: 2rem;
+
+    img {
+      padding: 0.5rem;
+    }
+  }
+`
+const LoadingWrapper = styled.div`
+  width: 100%;
+  height: 100%;
+  align-items: center;
+  justify-content: center;
+  display: flex;
+  flex-direction: column;
+`
 
 const NftAsset: React.FC<Props> = ({ defaultEthAddr, nftsActions, ...rest }) => {
   const { contract, id } = rest.computedMatch.params
-  console.log(contract)
-  const [results] = useAssetQuery({
+  // @ts-ignore
+  const [asset] = useAssetQuery({
     variables: { filter: { contract_address: contract, token_id: id } }
   })
-
-  return <div>{JSON.stringify(results)}</div>
+  const [assets] = useAssetsQuery({
+    variables: { filter: { contract_address: contract } }
+  })
+  const [Tab, setTab] = useState('details')
+  const NULL_ADDRESS = '0x0000000000000000000000000000000000000000'
```

Comment:
NULL_ADDRESS already defined in @core package somewhere. Import from there.

const WETH_ADDRESS = window.coins.WETH.type.erc20Address